src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -612,13 +548,13 @@ private void removeTerritoriesThatArentWorthAttacking(final List<ProTerritory> p
         int minCost = Integer.MAX_VALUE;
         Unit minUnit = null;
         for (final Unit u : t.getUnits().getMatches(Matches.unitIsOwnedBy(player))) {
-          if (playerCostMap.getInt(u.getType()) < minCost) {
-            minCost = playerCostMap.getInt(u.getType());
+          if (ProData.unitValueMap.getInt(u.getType()) < minCost) {
```

Comment:
A good place to take the same approach as you did earlier,
 `+          if (territoryManager.haveUsedAllAttackTransports()) {`
Thus essentially the suggestion is a new method on ProData that takes the "unit" and "minCost" as parameters and returns this boolean value for you.